internal/bminventory/inventory.go

```diff
@@ -562,6 +565,19 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
 	}
 
+	if b.ocmClient != nil {
+		log.Infof("Creating AMS subscription for cluster %v", id)
+		sub, err := b.ocmClient.AccountsMgmt.CreateSubscription(ctx, id)
+		if err != nil {
+			return common.NewApiError(http.StatusInternalServerError, err)
+		}
+		log.Infof("Patching DB with AMS subscription ID for cluster %v", id)
+		dbReply := b.db.Model(&common.Cluster{}).Where("id = ?", cluster.ID.String()).Updates(map[string]interface{}{"ams_subscription_id": sub.ID()})
```

Comment:
Not sure about this comment, do you think that it will be better to replace "ams_subscription_id" to more generic field name?

Reply:
Don't know what comment you are referring to @ErezAlster